src/classes/Addresses.cls

```diff
@@ -0,0 +1,1196 @@
+/*
```

Comment:
This class is just too large and a mixed collection of private, public, instance and static methods; some of which are quite long. I think it's going to be necessary to break this class down into something that is manageable - where each is focused on a specific behavior. My main concern with this at the moment is that it doesn't necessarily implement much from a "Domain" perspective that addresses flow and logic - especially around making it easier to make adjustments to the logic.

Reply:
Yep, this is definitely still a work in progress.  Migrating all of the logic out of the two ADDR_ TDTM classes AND fully refactoring every extracted/moved method proved to be too much to fit in at once.  The code is currently in a state where it has been extracted and moved, but not yet fully factored down.  As this class continues to be factored down, more logic gets pushed down into the NPSP_Address entity and more methods get extracted and de-duped in this class.  This process is underway, but not complete.   See the video I posted in the Hummingbird Slack channel and this commit that shows logic continue to get pushed down into the entity: [551677c](https://github.com/SalesforceFoundation/NPSP/pull/6330/commits/551677c6d335dced8e7974ee176986a62f693153)